### DISCUSSION TITLE
Mediawiki Setup

### DISCUSSION THREAD

**Discussion by alephate** [evennia/evennia#2593] [Community Contribs & Snippets]
# Mediawiki Instructions for 2021

So you want to set up mediawiki for your project, and you already have
Evennia running on a digital ocean Ubuntu 20.4 droplet, with an NginX proxy
set up and configured to use your domain for Evennia. 

## The Basic Idea

As the mediawiki installation docs say, the first thing you need is to
set up a LAMP stack on your droplet. This is Apache, PHP, and MySQL. You
can do this concurrently with everything else that you have set up, just
ssh into your droplet server and navigate to the root directory and install
everything from there. Make Apache listen on port 8080 so it doesn't
interfere with NginX and all is good.

## Start by Installing Mediawiki

Make sure everything is updated:

`sudo apt update`

Set up the LAMP stack:

`sudo apt-get install apache2 mysql-server php php-mysql libapache2-mod-php php-xml php-mbstring`

Now when you first set up apache2, it will give an error and not run 
because it clashes on port 80 with NginX. So you have to go into its
port configuration file and set it up to listen on port 8080.

'sudo nano /etc/apache2/ports.conf'

Edit the file to listen on port 8080, close and save. Then try to 
start Apache again:

`sudo service apache2 start`

From there you can pick up from this tutorial to finish installing 
mediawiki:

https://www.mediawiki.org/wiki/Manual:Running_MediaWiki_on_Debian_or_Ubuntu

## Accessing Your Apache Server Past Evennia

Now you have mediawiki presumably set up on your server, but you 
can't look at it because your evennia setup is getting in the way.

Use this tutorial to fix that, so you can access the mediawiki 
config page. Only use the tutorial up to step 5, you don't need
step 5 and onward.

Also, there is one outdated part of this tutorial that we will
fix in the end.

https://www.digitalocean.com/community/tutorials/how-to-configure-nginx-as-a-web-server-and-reverse-proxy-for-apache-on-one-ubuntu-18-04-server

Now to fix the outdated part:

`sudo nano /etc/apache2/mods-enabled/fastcgi.conf`

Update the version number in there to 7.4 and you're set! :)

So it should look like this:

```
<IfModule mod_fastcgi.c>
  AddHandler fastcgi-script .fcgi
  FastCgiIpcDir /var/lib/apache2/fastcgi
  AddType application/x-httpd-fastphp .php
  Action application/x-httpd-fastphp /php-fcgi
  Alias /php-fcgi /usr/lib/cgi-bin/php-fcgi
  FastCgiExternalServer /usr/lib/cgi-bin/php-fcgi -socket /run/php/php7.4-fpm.sock -pass-header Authorization
  <Directory /usr/lib/cgi-bin>
    Require all granted
  </Directory>
</IfModule>
```

Now you can open this link to configure your mediawiki:

http://yourdomain.com:8080/mediawiki

When you are done configuring, it will send LocalSettings.php
to your machine, and you will have to transfer that to your
remote set up in your preferred way and then move it to the
right place.

`sudo mv ~/Downloads/LocalSettings.php /var/lib/mediawiki/`

**Comment by jarenko** [comment]
Hello @alephate. I completed this successfully. I have only a couple points of feedback!

1. You may consider moving the line `Only use the tutorial up to step 5, you don't need
step 5 and onward.` to above the external tutorial link. In my case I went in order and clicked the link and was working on the tutorial from another tab and someone may do more than is necessary.
2. This might just be a me problem, but the tutorial has you verify the mediawiki is working by going to `yourdomain.com/mediawik`i, but I had not thought to include my Apache port so in my case it is was `yourdomain.com:8080/mediawiki`. Your tutorial does call this out, but I did not realize my mistake until I left the mediawiki tutorial and returned to yours!

These are the points I got stuck on -- other than that it was very easy to follow!
